lib/ClaimVerifier.ts

```diff
@@ -33,44 +30,46 @@ export class ClaimVerifier {
   }
 
   /**
-   * @description Checks that claim represents role and was
+   * @description Checks that credential represents role and was
    * issued by issuer required by role definition
    *
-   * @param claim: role claim
+   * @param credential: Role EIP191Jwt
    */
-  private async verifyRole(claim: Required<OffchainClaim>): Promise<{
-    name: IRoleDefinition['roleName'];
-    namespace: OffchainClaim['claimType'];
+  private async verifyRole(credential: Required<RoleEIP191JWT>): Promise<{
+    name: IRoleDefinitionV2['roleName'];
+    namespace: RoleEIP191JWT['payload']['claimData']['claimType'];
   } | null> {
     // if (!(await this.verifySignature(claim))) {
     //   return null;
     // }
 
-    const role = await this.getRoleDefinition(claim.claimType);
-    if (!role) {
-      return null;
-    }
-
-    if (role.version !== claim.claimTypeVersion) {
+    if (!credential?.payload) {
+      Logger.info('RolePayload does not exist');
       return null;
     }
+    if (credential.payload) {
+      const role = await this.getRoleDefinition(
+        credential.payload.claimData.claimType
+      );
+      if (!role) {
+        Logger.info('Role does not exist');
+        return null;
+      }
 
-    if (
-      role.issuer?.issuerType === 'DID' &&
-      Array.isArray(role.issuer?.did) &&
-      role.issuer?.did.includes(claim.iss)
-    ) {
-      return {
-        name: role.roleName,
-        namespace: claim.claimType,
-      };
-    } else if (role.issuer?.issuerType === 'Role' && role.issuer.roleName) {
-      const issuerClaims = await this.getOffchainClaims(claim.iss);
-      const issuerRoles = issuerClaims.map((claim) => claim.claimType);
-      if (issuerRoles.includes(role.issuer.roleName)) {
+      if (role.version !== credential.payload.claimData.claimTypeVersion) {
```

Comment:
I think this is the right place for the check, even if we move this to `getRoleDefinition` then also it will be a call inside it to validate and would run every time `getRoleDefinition` is called and also it just relies on namespace this way.

Reply:
The purpose of moving this check in `getRoleDefinition` is to avoid repeated checks

Reply:
I'm not really following. You call `getRoleDefinition` once for one credential, so even if we move this validation inside of `getRoleDefinition` then also it would be performed once for that credential. How this will avoid repeated calls? In both case this validation will happen the same number of times the `getRoleDefinition` is called. Am I missing something?

Reply:
`getRoleDefinition` is a general purpose function which fetches the roleDefinition based on namespace. So it may or mayn't have use case to validate the roleDefinition against the `claimTypeVersion`. So IMO it is not a mandatory check.